clients/docs/java.rst

```diff
@@ -0,0 +1,320 @@
+
+
+Prerequisites
+-------------
+
+Client
+~~~~~~
+
+-  Java 1.8 or higher to run the demo application.
+
+-  Maven to compile the demo application.
+
+
+Kafka Cluster
+~~~~~~~~~~~~~
+
+.. include:: includes/client-example-prerequisites.rst
+
+
+Setup
+-----
+
+#. Clone the `confluentinc/examples GitHub repository
+   <https://github.com/confluentinc/examples>`__ and check out the
+   :litwithvars:`|release|-post` branch.
+
+   .. codewithvars:: bash
+
+      git clone https://github.com/confluentinc/examples
+      cd examples
+      git checkout |release|-post
+
+#. Change directory to the example for Java.
+
+   .. code-block:: bash
+
+      cd clients/cloud/java/
+
+#. .. include:: includes/client-example-create-file.rst
+
+
+Basic Producer and Consumer
+---------------------------
+
+In this example, the producer writes Kafka data to a topic in your Kafka
+cluster. Each record has a key representing a username (for example, ``alice``)
+and a value of a count, in JSON format (for example, ``{"Count": 0}``). The
+consumer reads the same topic and keeps a rolling sum of the counts as it
+processes each record.
+
+
+Produce Records
+~~~~~~~~~~~~~~~
+
+#. Run the producer, passing in arguments for:
+
+   - the local file with configuration parameters to connect to your Kafka cluster
+   - topic name
+
+   .. code-block:: text
+
+      # Compile the Java code
+      mvn clean package
+
+      # Run the producer
+          # If the topic does not already exist, the code will use the Kafka Admin Client API to create the topic
+      mvn exec:java -Dexec.mainClass="io.confluent.examples.clients.cloud.ProducerExample" \
+      -Dexec.args="$HOME/.confluent/java.config test1"
+
+#. Verify that the producer received all the messages. You should see:
+
+   .. code-block:: bash
+
+      ...
+      Producing record: alice {"count":0}
+      Producing record: alice {"count":1}
+      Producing record: alice {"count":2}
+      Producing record: alice {"count":3}
+      Producing record: alice {"count":4}
+      Producing record: alice {"count":5}
+      Producing record: alice {"count":6}
+      Producing record: alice {"count":7}
+      Producing record: alice {"count":8}
+      Producing record: alice {"count":9}
+      Produced record to topic test1 partition [0] @ offset 0
+      Produced record to topic test1 partition [0] @ offset 1
+      Produced record to topic test1 partition [0] @ offset 2
+      Produced record to topic test1 partition [0] @ offset 3
+      Produced record to topic test1 partition [0] @ offset 4
+      Produced record to topic test1 partition [0] @ offset 5
+      Produced record to topic test1 partition [0] @ offset 6
+      Produced record to topic test1 partition [0] @ offset 7
+      Produced record to topic test1 partition [0] @ offset 8
+      Produced record to topic test1 partition [0] @ offset 9
+      10 messages were produced to topic test1
+      ...
+
+#. View the :devx-examples:`producer
```

Comment:
@tikimims Unnecessary line break with below?

Reply:
Oh sorry —just a styling preference on my end. I can remove that. :-)

Reply:
Updated